msal/src/Microsoft.Identity.Client/Internal/Requests/AuthenticationRequestParameters.cs

```diff
@@ -117,6 +118,30 @@ internal class AuthenticationRequestParameters
             return parameters;
         }
 
+#if DESKTOP || NETSTANDARD1_3
+        public override bool Equals(object obj)
+        {
+            if (obj == null)
+            {
+                throw new ArgumentNullException("credential");
+            }
+            else if (obj is ClientCredential credential)
+            {
+                return credential.Audience == Authority.SelfSignedJwtAudience && credential.ContainsX5C == SendCertificate;
+            }
+            else
+                throw new ArgumentException("Parameter obj is not of type " + typeof(ClientCredential).ToString());
+        }
+
+        public override int GetHashCode()
+        {
+            int hash = 13;
+            hash *= hash + 31 + Authority.SelfSignedJwtAudience.GetHashCode();
+            hash *= hash + 31 + SendCertificate.GetHashCode();
+            return hash;
+        }
+#endif
```

Comment:
this is not right. It's not the AuthenticationRequestParameter we should be looking at. It's the assertions we need to validate for equality.

Reply:
#Resolved